src/dir.cr

```diff
@@ -101,7 +96,7 @@ class Dir
   end
 
   def each_child
-    ChildIterator.new(self)
+    @dir.reject { |x| {".", ".."}.includes?(x) }
```

Comment:
I'm not sure if/why this is preferable over a custom iterator class?

Reply:
It's easier to read, and a custom iterator would have to wrap `@dir` regardless. So comparing this to a custom iterator, the only overhead would be a single function pointer call overhead. Given that every call to `next` will perform a syscall (which busts cache and takes 200+ cycles!): this is a very acceptable overhead.